src/savegame.cpp

```diff
@@ -1000,8 +1000,33 @@ void overmap::unserialize( std::istream &fin )
                 }
                 npcs.push_back( new_npc );
             }
+        } else if( name == "map_extra_triggers" ) {
+            jsin.start_array();
+            while( !jsin.end_array() ) {
+                jsin.start_object();
+                map_extra_trigger trigger;
+                while( !jsin.end_object() ) {
+                    std::string member_name = jsin.get_member_name();
+                    if( member_name == "omt_pos_1" ) {
+                        jsin.read( trigger.omt_pos_1 );
+                    } else if( member_name == "size" ) {
+                        jsin.read( trigger.size );
+                    } else if( member_name == "trigger_distance" ) {
+                        jsin.read( trigger.trigger_distance );
+                    } else if( member_name == "map_special" ) {
+                        jsin.read( trigger.map_special );
+                    } else if( member_name == "triggered" ) {
+                        jsin.read( trigger.triggered );
+                    }
+                }
+                map_extra_triggers.push_back( trigger );
+            }
         }
     }
+    if(map_extra_triggers.size() == 0)
```

Comment:
Why are you checking the size when you actually want to know whether it's `empty` or not?